src/constructors/createGlobalStyle.js

```diff
@@ -0,0 +1,62 @@
+// @flow
+import React from 'react'
+import PropTypes from 'prop-types'
+import { CONTEXT_KEY, STATIC_EXECUTION_CONTEXT } from '../constants'
+import _GlobalStyle from '../models/GlobalStyle'
+import StyleSheet from '../models/StyleSheet'
+import ServerStyleSheet from '../models/ServerStyleSheet'
+import type { CSSConstructor, Interpolation, Stringifier } from '../types'
+import withTheme from '../hoc/withTheme'
+import hashStr from '../vendor/glamor/hash'
+
+export default (stringifyRules: Stringifier, css: CSSConstructor) => {
+  const GlobalStyle = _GlobalStyle(stringifyRules)
+
+  const createGlobalStyle = (
+    strings: Array<string>,
+    ...interpolations: Array<Interpolation>
+  ) => {
+    const rules = css(strings, ...interpolations)
+    const id = `sc-global-${hashStr(JSON.stringify(rules))}`
+    const style = new GlobalStyle(rules, id)
+
+    class GlobalStyleComponent extends React.Component {
+      static contextTypes = {
+        [CONTEXT_KEY]: PropTypes.oneOfType([
+          PropTypes.instanceOf(StyleSheet),
+          PropTypes.instanceOf(ServerStyleSheet),
+        ]),
+      }
+
+      static styledComponentId = id
+
+      componentWillMount() {
+        const { props } = this
+        const sheet = this.context[CONTEXT_KEY] || StyleSheet.master
+        const context = style.isStatic ? STATIC_EXECUTION_CONTEXT : { ...props }
```

Comment:
Theming logic is not in place yet? We could mark this as a WIP since it’s been a while and merge this first?

Reply:
@kitten I'm all for merging this now rather than later…

Reply:
Currently this uses `withTheme` - can we go forward with the current solution and replace it with a util-based one later?

Reply:
Added a comment noting the `withTheme` usage should be removed when theming is refactored